### DISCUSSION TITLE
RFC OpenConnection Exception in Azure Function app

### DISCUSSION THREAD

**Discussion by hemanthaar** [dbosoft/YaNco#184] [Q&A]
Hi Frank,
Hope you are doing well.

While deploying the function app to Azure, we are getting below exception at .net runtime . The dll files are copied to the path and also the function app is deployed in our virtual network which can connect to our sap system.

Could you please let me know where I am going wrong. Thanks in advance.

![image](https://user-images.githubusercontent.com/74851652/172446078-e11f33dd-5181-4ab4-8ddc-47ad4804a7f0.png)

P.S: The function app works perfectly when running from my local pc.

Regards
Hemanth

**Comment by fw2568** [comment]
Hi,

first: I can only repeat my advise not to use SAP connections in Azure functions, Azure functions are not designed for complex applications - see https://github.com/dbosoft/SAPHub for a reference architecture in cloud scenarios. 

It looks like a issue with the Azure App Sandbox and something the **saprfcsdk** is trying to do (for example access to filesystem). So that is out of scope for this library.

If you still want to use Azure functions I would give Linux a try (however, this has also many pitfalls for native code) or to open a ticket in Azure support to analyze the cause in detail. 

Best Regards,
Frank
